src/coreclr/jit/llvmcodegen.cpp

```diff
@@ -880,7 +880,7 @@ void Llvm::buildStoreLocalVar(GenTreeLclVar* lclVar)
 
     if (lclVar->HasSsaName())
     {
-        _localsMap.Set({lclNum, lclVar->GetSsaNum()}, localValue);
+        _localsMap.Set({lclNum, lclVar->GetSsaNum()}, localValue, JitHashTable<Compiler::SSAName, Compiler::SSAName, Value*>::Overwrite);
```

Comment:
This is not right. It indicates we are visiting the same SSA definition twice, which is not legal. What is the cause?

Reply:
Dump attached `Object__GetHashCode`
[dump.txt](https://github.com/dotnet/runtimelab/files/10376220/dump.txt)

What happens is that the prolog generation for 
```
Setting V02's initial value to uninit
  %2 = freeze i32 undef
```
Sets lcl/ssa 2:1
And then 
```
Generating: N004 ( 24,  8) [000005] DA-XG-------              *  STORE_LCL_VAR int    V02 tmp0         d:1
```
fails setting the same key.  Is `lvaInSsa` returning true now in the prolog and it wasn't before?  I could check that.

Reply:
No need... The bug is here:
```
unsigned ssaNum = varDsc->lvPerSsaData.AllocSsaNum(m_allocator, block,
                                                   defNode->OperIs(GT_ASG) ? defNode->AsOp() : nullptr);
```
It needs to record the SSA def node for `STORE_LCL_VAR` as well. Flipping the condition to defNode->OperIs(GT_ASG) ? defNode->AsOp() : nullptr)` to `!defNode->OperIs(GT_CALL) ? defNode->AsOp() : nullptr)` should achieve this.

Reply:
I see, should have been no need for initialisation. Thanks